tests/compiler/std/string.optimized.wat

```diff
@@ -2402,83 +2397,108 @@
   (local $1 i32)
   (local $2 i32)
   (local $3 i32)
+  (local $4 i32)
   i32.const 552
   call $~lib/rt/pure/__retain
-  drop
+  local.tee $1
+  i32.eqz
+  if
+   i32.const 576
+   local.set $2
+   local.get $1
+   i32.const 576
+   i32.ne
+   if
+    i32.const 576
+    call $~lib/rt/pure/__retain
+    local.set $2
+    local.get $1
+    call $~lib/rt/pure/__release
+   end
+   local.get $2
+   local.set $1
+  end
```

Comment:
Hmm. Is it ok? It seems a lot of extra code

Reply:
Overall the PR adds exactly one line, so I guess that what's being added here has been removed somewhere else. Quite some code motion :)

Reply:
"string.optimized.wat" added 10 extra lines. But I guess it's ok.

Reply:
I wonder if this is a shortcoming of Binaryen's `LocalGraph` not having sets for function arguments. In the code, it is mentioned that gets affected by such sets have a set of `null`, but I haven't figured out so far how I would properly traverse the graph beginning at a function argument, especially when flatten is run and a function argument is copied into temp locals that are not influenced by anything. That's most likely just me not knowing enough about it, but mentioning in case @kripken sees something I don't see :)

Reply:
LocalGraph should have all the possible information for those - when you look at the sets affecting a get, they are either actual sets in the function body, or the initial value, which is either zero or the parameter value. The initial value doesn't have a set, so we indicate those by a null pointer.

This shouldn't inhibit any optimizations AFAIK.